lib/message_bus/codec/oj_fast.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'oj'
+
+module MessageBus
+  module Codec
+    class FastIdList
+      def self.from_array(array)
+        new(array.sort.pack("V*"))
+      end
+
+      def self.from_string(string)
+        new(string)
+      end
+
+      def initialize(packed)
+        @packed = packed
+      end
+
+      def include?(id)
+        found = (0...length).bsearch do |index|
+          start = index * 4
+          @packed[start, start + 4].unpack1("V") >= id
```

Comment:
Hum, have you tried `@packed.byteslice(start..-1).unpack1("V")`, that should return you a shared string so avoid copy. 

But the current code only copy 4 bytes, so maybe it makes no difference.

Reply:
Thanks @casperisfine, I added this in! also fixed the terrible bug in that line